homeassistant/components/drop_connect/coordinator.py

```diff
@@ -0,0 +1,66 @@
+"""DROP device data update coordinator object."""
+from __future__ import annotations
+
+from collections.abc import Callable
+import logging
+from typing import TYPE_CHECKING
+
+import attr
+from dropmqttapi.mqttapi import DropAPI
+
+from homeassistant.components import mqtt
+from homeassistant.components.mqtt import ReceiveMessage
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import CONF_DATA_TOPIC, CONF_DEVICE_TYPE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class DROPDeviceDataUpdateCoordinator(DataUpdateCoordinator):
+    """DROP device object."""
+
+    _drop_api: DropAPI | None = None
+    _unsubscribe_callback: Callable[[], None] | None = attr.ib()
+
+    def __init__(self, hass: HomeAssistant, unique_id: str) -> None:
+        """Initialize the device."""
+        super().__init__(hass, _LOGGER, name=f"{DOMAIN}-{unique_id}")
+        if TYPE_CHECKING:
+            assert self.config_entry is not None
+        self._drop_api = DropAPI()
```

Comment:
Does the construction of the DropAPI have any side effects?

Reply:
```suggestion
        self.drop_api = DropAPI()
```

Reply:
The construction of the DropAPI object only returns a reference to the API object.  

Renaming self._drop_api and removing the property creates a series of cascading problems.  The end result is that it can be None, so mypy complains about every endpoint referenced in sensors.py as being a missing attribute.  Perhaps I'm not understanding the suggested change...

Reply:
What I would expect is that you set a class property in the constructor, so this means the class property is not None.

I don't see how it can be None

Reply:
I tried the suggestions, and it seems to be just fine. When can `self.drop_api` be None?

Reply:
It can't be None.  The problems I was having were my own doing.  Sometimes it really helps just to isolate changes in their own commits...  Fixed in 599e58c2197c2d54bbd190000dfea160ec57e87e.